c7n/resources/workspaces.py

```diff
@@ -519,6 +519,44 @@ def process(self, resources, event=None):
         return results
 
 
+@WorkspacesWeb.filter_registry.register('user-settings')
+class UserSettingsFilter(ValueFilter):
+    """
+    Filters workspaces secured browsers based on their user settings.
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: user-settings-match
+                resource: workspaces-web
+                filters:
+                  - type: user-settings
+                    key: copyAllowed
+                    value: Disabled
+    """
+
+    schema = type_schema('user-settings', rinherit=ValueFilter.schema)
+    schema_alias = False
+    permissions = ('workspaces-web:GetUserSettings',)
+    policy_annotation = "c7n:UserSettings"
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('workspaces-web')
+        results = []
+        for r in resources:
+            if (self.policy_annotation not in r) and ('userSettingsArn' in r):
+                r[self.policy_annotation] = self.manager.retry(
+                    client.get_user_settings,
+                    userSettingsArn=r['userSettingsArn']).get(
+                        'userSettings', {})
+            results.append(r)
```

Comment:
technically we don't really need the separate results list, as resources is being modified in place.